accumulator_updater/programs/accumulator_updater/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+mod macros;
+
+use anchor_lang::{
+    prelude::*,
+    solana_program::{
+        program::invoke_signed,
+        system_instruction,
+        sysvar::{
+            self,
+            instructions::get_instruction_relative,
+        },
+    },
+};
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+
+#[program]
+pub mod accumulator_updater {
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let whitelist = &mut ctx.accounts.whitelist;
+        whitelist.bump = *ctx.bumps.get("whitelist").unwrap();
+        Ok(())
+    }
+
+    pub fn add_allowed_program(
+        ctx: Context<AddAllowedProgram>,
+        allowed_program: Pubkey,
+    ) -> Result<()> {
+        let whitelist = &mut ctx.accounts.whitelist;
+        require_keys_neq!(allowed_program, Pubkey::default());
+        require!(
+            !whitelist.allowed_programs.contains(&allowed_program),
+            AccumulatorUpdaterError::DuplicateAllowedProgram
+        );
+        whitelist.allowed_programs.push(allowed_program);
+        Ok(())
+    }
+
+    /// Add new account(s) to be included in the accumulator
+    pub fn add_account<'info>(
+        ctx: Context<'_, '_, '_, 'info, AddAccount<'info>>,
+        base_account: Pubkey,
+        data: Vec<Vec<u8>>,
+        account_type: u32,
```

Comment:
originally i was thinking that the serialization format would be uniquely identified through a tuple of `(account_type, account_schema)`. Based on how pyth is currently structured, each ix only works on one type of `account_type` at a time. this can be changed though.

Reply:
the flow i was thinking was this:

1. pyth-oracle invokes `add_price` which creates a `pythPriceAccount`
2. lets say there's 2 different schemas we want for `PriceAccount` - "full" & "compact", the pyth-oracle will transform the original `pythPriceAccount` data  - 
```
let accumulator_inputs: Vec<Vec<u8>> = [Schema::Full, Schema::compact].map(|s| transform(AccountType::Price, s, pythPriceAccount).serialize())
```
3. as part of `add_price`, pyth-oracle invokes a cpi call to accumulator-udpater to generate these `AccumulatorInput` accounts

the final key for the accumulator can be just the AccumulatorInput PDA address since right now the PDA `seeds = [caller-program(pyth-oracle), base-account(pythPriceAccount.key()), schema]`